concordium-consensus/src/Concordium/Scheduler/EnvironmentImplementation.hs

```diff
@@ -55,32 +59,36 @@ class CanExtend (TransactionLog a) => HasSchedulerState a where
   schedulerExecutionCosts :: Lens' a Amount
   schedulerTransactionLog :: Lens' a (TransactionLog a)
   nextIndex :: Lens' a TransactionIndex
+  transactionVerificationCache :: Lens' a TransactionVerificationCache
 
 data NoLogSchedulerState (m :: DK.Type -> DK.Type)= NoLogSchedulerState {
   _ssBlockState :: !(UpdatableBlockState m),
   _ssSchedulerEnergyUsed :: !Energy,
   _ssSchedulerExecutionCosts :: !Amount,
-  _ssNextIndex :: !TransactionIndex
+  _ssNextIndex :: !TransactionIndex,
+  _ssTransactionVerificationCache :: !TransactionVerificationCache
   }
 
 mkInitialSS :: UpdatableBlockState m -> NoLogSchedulerState m
 mkInitialSS _ssBlockState = NoLogSchedulerState{
     _ssSchedulerEnergyUsed = 0,
     _ssSchedulerExecutionCosts = 0,
     _ssNextIndex = 0,
+    _ssTransactionVerificationCache = Map.empty,
     ..
   }
 
 makeLenses ''NoLogSchedulerState
 
-instance HasSchedulerState (NoLogSchedulerState m) where
-  type SS (NoLogSchedulerState m) = UpdatableBlockState m
-  type TransactionLog (NoLogSchedulerState m) = ()
-  schedulerBlockState = ssBlockState
-  schedulerEnergyUsed = ssSchedulerEnergyUsed
-  schedulerExecutionCosts = ssSchedulerExecutionCosts
-  nextIndex = ssNextIndex
+instance HasSchedulerState (NoLogSchedulerState m, TransactionVerificationCache) where
+  type SS (NoLogSchedulerState m, TransactionVerificationCache) = UpdatableBlockState m
+  type TransactionLog (NoLogSchedulerState m, TransactionVerificationCache) = ()
+  schedulerBlockState = _1 . ssBlockState
+  schedulerEnergyUsed = _1 . ssSchedulerEnergyUsed
+  schedulerExecutionCosts = _1 . ssSchedulerExecutionCosts
+  nextIndex = _1 . ssNextIndex
   schedulerTransactionLog f s = s <$ f ()
+  transactionVerificationCache = _2
```

Comment:
This seems very confusing to me. Why use a pair instead of just using the `ssTransactionVerificationCache` component of `NoLogSchedulerState`?